### ISSUE TITLE
Nested parametrize_with_cases does not collect test if fixture is used multiple times

### ISSUE THREAD

**Issue by reyreaud-l** [smarie/python-pytest-cases#179]
Hello there!

First of all thanks for that package for it is very awesome!

I was working on doing some weird matrix testing to test permissions of user against other users resources. In order to do that I had large cases classes that boiled down to the minimum example below:

```
import pytest_cases as pytest


@pytest.fixture
def db_dep():
    return None

class CaseX:
    def case_one(self, db_dep):
        return 1

    def case_two(self, db_dep):
        return 2


class CaseY:
    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_one(self,db_dep,x):
        return x, 1

    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_two(self,db_dep,x):
        return x, 1


@pytest.parametrize_with_cases("x,y", cases=CaseY)
def test_nested_parametrize(x, y):
    pass
``` 

I know the example could be simplified, but in my use case, CaseX's case need to have db_access to create a user, then CaseY get this user, and create a new one (so they need db access). Then the test received the matrix of created users.

Here when you run pytest, the following is collected:
```
Test session starts (platform: linux, Python 3.9.1, pytest 6.2.1, pytest-sugar 0.9.4)
rootdir: /home/kexo/Projects/trajaan/backend
plugins: mock-3.5.1, cov-2.10.1, xdist-2.2.0, forked-1.3.0, cases-3.1.1, sugar-0.9.4
collecting ... 
<Module test.py>
  <Function test_nested_parametrize[x_two-one]>
  <Function test_nested_parametrize[x_two-two]>

```

I can manage to get the correct collection only if I remove the fixture dependency on ClassX:
```
import pytest_cases as pytest


@pytest.fixture
def db_dep():
    return None

class CaseX:
    def case_one(self):
        return 1

    def case_two(self):
        return 2


class CaseY:
    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_one(self,db_dep,x):
        return x, 1

    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_two(self,db_dep,x):
        return x, 1


@pytest.parametrize_with_cases("x,y", cases=CaseY)
def test_nested_parametrize(x, y):
    pass
```

```
Test session starts (platform: linux, Python 3.9.1, pytest 6.2.1, pytest-sugar 0.9.4)
rootdir: /home/kexo/Projects/trajaan/backend
plugins: mock-3.5.1, cov-2.10.1, xdist-2.2.0, forked-1.3.0, cases-3.1.1, sugar-0.9.4
collecting ... 
<Module test.py>
  <Function test_nested_parametrize[x_one-one]>
  <Function test_nested_parametrize[x_one-two]>
  <Function test_nested_parametrize[x_two-one]>
  <Function test_nested_parametrize[x_two-two]>


Results (0.02s):
```

Is it expected ? How can work around that limitation ? I tried defining another fixture as `db_dep` and naming it something else but the limitation is still here:
```
import pytest_cases as pytest


@pytest.fixture
def db_dep():
    return None

@pytest.fixture
def db_dep2():
    return None

class CaseX:
    def case_one(self, db_dep2):
        return 1

    def case_two(self, db_dep2):
        return 2


class CaseY:
    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_one(self,db_dep,x):
        return x, 1

    @pytest.parametrize_with_cases("x", cases=CaseX)
    def case_x_two(self,db_dep,x):
        return x, 1


@pytest.parametrize_with_cases("x,y", cases=CaseY)
def test_nested_parametrize(x, y):
    pass
```
```
Test session starts (platform: linux, Python 3.9.1, pytest 6.2.1, pytest-sugar 0.9.4)
rootdir: /home/kexo/Projects/trajaan/backend
plugins: mock-3.5.1, cov-2.10.1, xdist-2.2.0, forked-1.3.0, cases-3.1.1, sugar-0.9.4
collecting ... 
<Module test.py>
  <Function test_nested_parametrize[x_two-one]>
  <Function test_nested_parametrize[x_two-two]>


Results (0.02s):
```

Thanks for having a look!

**Comment by smarie** [comment]
Thanks @reyreaud-l for posting this ! Indeed this is strange, I'll need to have a closer look.

As a workaround, you can create a fixture for x, it seems to work fine:

```python
from pytest_cases import fixture, parametrize_with_cases

@fixture
def db_dep():
    return None

class CaseX:
    def case_one(self, db_dep):
        return 1

    def case_two(self, db_dep):
        return 2

@fixture
@parametrize_with_cases("xc", cases=CaseX)
def x(xc):
    return xc

class CaseY:
    def case_x_one(self,db_dep,x):
        return x, 1

    def case_x_two(self,db_dep,x):
        return x, 1

@parametrize_with_cases("x,y", cases=CaseY)
def test_nested_parametrize(x, y):
    pass
```

**Comment by smarie** [comment]
I found the issue: the `<func>.place_as` magic attribute supported by pytest and set by `parametrize_with_cases` was not supported by `parametrize_with_cases` (in this nesting scenario). Therefore the two case functions in CaseY were assigned with the same line number (1) and would override each other.

I fixed this, and in addition added a safeguard ensuring that no function would be silently dropped because another gets the same line number.
